pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,309 @@
+		o.Image = "quay.io/openshift/origin-must-gather:v4.0"
+	}
+	o.PrinterCreated, err = printers.NewTypeSetter(scheme.Scheme).WrapToPrinter(&printers.NamePrinter{Operation: "created"}, nil)
+	if err != nil {
+		return err
+	}
+	o.RsyncRshCmd = rsync.DefaultRsyncRemoteShellToUse(cmd.Parent())
+	return nil
+}
+
+type MustGatherOptions struct {
+	genericclioptions.IOStreams
+
+	Config *rest.Config
+	Client kubernetes.Interface
+
+	NodeName string
+	DestDir  string
+	Image    string
+	Command  []string
+	Keep     bool
+
+	RsyncRshCmd string
+
+	PrinterCreated printers.ResourcePrinter
+}
+
+// Run creates and runs a must-gather pod.d
+func (o *MustGatherOptions) Run(rsyncCmd *cobra.Command) error {
+	if len(o.Image) == 0 {
+		return fmt.Errorf("missing an image")
+	}
+
+	var err error
+
+	// create namespace
+	ns, err := o.Client.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "openshift-must-gather-",
+			Labels: map[string]string{
+				"openshift.io/run-level": "0",
+			},
+			Annotations: map[string]string{
+				"oc.openshift.io/command": "oc adm must-gather",
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if !o.Keep {
+		defer func() {
+			err = o.Client.CoreV1().Namespaces().Delete(ns.Name, nil)
+		}()
+	} else {
+		o.PrinterCreated.PrintObj(ns, o.Out)
+	}
+
+	clusterRoleBinding, err := o.Client.RbacV1().ClusterRoleBindings().Create(o.newClusterRoleBinding(ns.Name))
+	if err != nil {
+		return err
+	}
+	if !o.Keep {
+		defer func() {
+			err = o.Client.RbacV1().ClusterRoleBindings().Delete(clusterRoleBinding.Name, &metav1.DeleteOptions{})
+		}()
+	} else {
+		o.PrinterCreated.PrintObj(clusterRoleBinding, o.Out)
+	}
+
+	// create pod
+	pod, err := o.Client.CoreV1().Pods(ns.Name).Create(o.newPod(o.NodeName))
+	if err != nil {
+		return err
+	}
+
+	// wait for pod to be running (gather has completed)
+	if err := o.waitForPodRunning(pod); err != nil {
+		return err
+	}
+
+	// copy the gathered files into the local destination dir
+	err = o.copyFilesFromPod(pod)
+	return err
+}
+
+func (o *MustGatherOptions) copyFilesFromPod(pod *corev1.Pod) error {
+	rsyncOptions := &rsync.RsyncOptions{
+		Namespace:     pod.Namespace,
+		Source:        &rsync.PathSpec{PodName: pod.Name, Path: "/must-gather/"},
+		ContainerName: "copy",
+		Destination:   &rsync.PathSpec{PodName: "", Path: o.DestDir},
+		Client:        o.Client,
+		Config:        o.Config,
+		RshCmd:        fmt.Sprintf("%s --namespace=%s", o.RsyncRshCmd, pod.Namespace),
+		IOStreams:     o.IOStreams,
+	}
+	rsyncOptions.Strategy = rsync.NewDefaultCopyStrategy(rsyncOptions)
+	return rsyncOptions.RunRsync()
+
+}
+
+func (o *MustGatherOptions) waitForPodRunning(pod *corev1.Pod) error {
+	phase := pod.Status.Phase
+	err := wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		var err error
+		if pod, err = o.Client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{}); err != nil {
+			klog.Error(err)
```

Comment:
Don't log the error here, if you care, log it at higher debug levels, it's not an error per se.

Reply:
removed in PR https://github.com/openshift/origin/pull/22561